test/doc/traits/callable_constant.cpp

```diff
@@ -0,0 +1,60 @@
+#include <iostream>
+#include <type_traits>
+#include <eve/traits/overload.hpp>
+#include <eve/wide.hpp>
+
+namespace eve
+{
+  // Constant generations functions have a precise behavior so they can be defined using the
+  // constant_callable scaffolding that enables this.
+  // Note that constant_callable provides support for conditional out of the box.
+  template<typename Options>
+  struct some_pi_t : constant_callable<some_pi_t, Options>
+  {
+    // A constant must provide a value() static member with up to two parameters:
+    //  * an eve::as instance to specify the output type
+    //  * an optional eve::options containing potential decorators passed to the constant
+    template<typename T>
+    static EVE_FORCEINLINE T value(eve::as<T> const&) { return static_cast<T>(3.14159216); }
+
+    // Supported function calls are still required
+    template<eve::floating_value T>
+    EVE_FORCEINLINE T     operator()(as<T>    v) const { return EVE_DISPATCH_CALL(v); }
+    EVE_FORCEINLINE auto  operator()(as<char> v) const { return EVE_DISPATCH_CALL(v); }
```

Comment:
We don't do auto! Otherwise the error messages are not as good.

Also - char for some_pi? Comment maybe like "this is purely example or smth"

Reply:
I am gonna make an arbitraryconsatnt instead, I need that to demonstrate that it can be non template